tests/valgrind-check/valgrind.sh

```diff
@@ -0,0 +1,193 @@
+#!/bin/bash
+
+function print_ps {
+    set +e
+    echo "CFEngine processes:"
+    ps aux | grep [c]f-
+
+    echo "Valgrind processes:"
+    ps aux | grep [v]algrind
+    set -e
+}
+
+function stop_daemons {
+    echo "Stopping cfengine daemons"
+    pkill -f cf-serverd  || echo "Did not kill cf-serverd"
+    pkill -f cf-execd    || echo "Did not kill cf-execd"
+    pkill -f cf-monitord || echo "Did not kill cf-monitord"
+}
+
+function no_errors {
+    set +e
+    grep -i "error" $1
+    grep -i "error" $1 && exit 1
+    set -e
+}
+
+function check_daemon_output {
+    echo "Examining $1:"
+    no_errors $1
+}
+
+function check_valgrind_output {
+    set -e
+    if [ ! -f "$1" ]; then
+        echo "$1 does not exists!"
+        exit 1
+    fi
+    echo "Looking for problems in $1:"
+    grep -i "ERROR SUMMARY: 0 errors" "$1"
+    cat $1 | sed -e "/ 0 errors/d" -e "/and suppressed error/d" -e "/a memory error detector/d" > filtered.txt
```

Comment:
You can use `sed ... < $1` instead of `cat|sed`.

Reply:
I dunno, I think mine is more readable for non-wizards 🤷